hawk/app/models/wizard.rb

```diff
@@ -1,203 +1,116 @@
+    # TODO: Check loaded
+    CrmScript.run ["verify", @name, params] do |item, err|
+      Rails.logger.debug "#{item}, #{err}"
+    end
   end
 
-  def script_file(name)
-    self.class.script_file(name)
+  def run(params)
+    # TODO: Check loaded
+    # TODO: live-update frontend
+    CrmScript.run ["run", @name, params] do |item, err|
+      Rails.logger.debug "#{item}, #{err}"
+    end
   end
 
   class << self
-    def parse(file)
-      REXML::Document.new(
-        file.read
-      ).tap do |xml|
-        name = xml.root.elements["shortdesc[@lang=\"#{I18n.locale.to_s.gsub("-", "_")}\"]|shortdesc[@lang=\"en\"]"].text.strip
-        description = xml.root.elements["longdesc[@lang=\"#{I18n.locale.to_s.gsub("-", "_")}\"]|longdesc[@lang=\"en\"]"].text.strip
-        order = xml.root.attributes["name"]
-
-        return Wizard.new(
-          id: file.basename(".xml").to_s,
-          order: order,
-          name: name,
-          description: description,
-          definition: file
-        )
-      end
+    def parse_brief(data)
+      return Wizard.new(
+               name: data['name'],
+               category: data['category'].strip.downcase,
+               shortdesc: data['shortdesc'].strip,
+               longdesc: data['longdesc'].gsub(/\n/, '<br>'),
+               loaded: false
+             )
     end
 
-    def find(id)
-      file = workflow_file(id).dup
-
-      if file
-        record = parse(file)
-        record.load!
-
-        record
-      else
-        raise CibObject::RecordNotFound, _("Requested workflow does not exist")
-      end
-    end
-
-    def all
-      workflow_files.values.map do |file|
-        parse(file)
-      end.sort_by(&:order)
-    end
-
-    def workflow_path
-      @workflow_path ||= Rails.root.join("config", "wizard", "workflows")
-    end
-
-    def workflow_files
-      @workflow_files ||= begin
-        files = workflow_path.children.select do |file|
-          if file.extname == ".xml"
-            file
-          end
-        end
-
-        {}.tap do |result|
-          files.each do |file|
-            result[file.basename(".xml").to_s] = file
-          end
-        end
-      end
+    def parse_full(data)
+      wizard = Wizard.new(
+        name: data['name'],
+        category: data['category'].strip.downcase,
+        shortdesc: data['shortdesc'].strip,
+        longdesc: data['longdesc'].gsub(/\n/, '<br>'),
+        loaded: false
+      )
+      wizard.load_from data
+      wizard
     end
 
-    def workflow_file(name)
-      workflow_files[name]
+    def find(name)
+      wizard = Wizard.all.select{|w| w.name == name}.first
+      raise CibObject::RecordNotFound, _("Requested wizard does not exist") if wizard.nil?
+      wizard.load!
+      wizard
     end
 
-    def template_path
-      @template_path ||= Rails.root.join("config", "wizard", "templates")
+    def exclude_wizard(item)
+      workflows = {'60-nfsserver' => true,
+                   'mariadb' => true,
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.